Jenkinsfile

```diff
@@ -1,40 +1,69 @@
 pipeline {
-  agent any
-  stages {
-    stage("Parallel Stage") {
-        parallel {
-            stage("linux build"){
-                agent { node { label 'linux' } }
-                steps {
-                        
-                    timeout( time: 180, unit: 'MINUTES' ) {
-                        withMaven(
-                              maven: 'maven3.6.1',
-                              publisherStrategy: 'EXPLICIT',
-                              options: [junitPublisher(disabled: false)],
-                              mavenLocalRepo: ".repository") {
-                            sh "mvn -V -B clean install -e -Dmaven.test.failure.ignore=true"
-                        }  
+    agent any
+    parameters {
+        booleanParam defaultValue: true, description: 'Build fails if coverage falls below value set in project property jacoco.coverage.target<br><li>True (default): Enables this behavior<li>False: Disables this behavior', name: 'failIfCoverageNotMet'
+    }
+    stages {
+        stage("Parallel Stage") {
+            parallel {
+                stage("linux build"){
+                    agent { node { label 'linux' } }
+                    steps {
+                        timeout( time: 180, unit: 'MINUTES' ) {
+                            withMaven(
+                                maven: 'maven3.6.1',
+                                publisherStrategy: 'EXPLICIT',
+                                options: [junitPublisher(disabled: false)],
+                                mavenLocalRepo: ".repository") {
+                                sh "mvn -Djacoco.haltOnFailure=${params.failIfCoverageNotMet} -V -B clean install -e -Dmaven.test.failure.ignore=true"
+                            }  
+                        }
+                    }    
+                }
+                stage("windows build"){
+                    agent { node { label 'windows' } }
+                    steps {
+                        timeout( time: 180, unit: 'MINUTES' ) {
+                            withMaven(
+                                maven: 'maven3.6.1',
+                                publisherStrategy: 'EXPLICIT',
+                                options: [junitPublisher(disabled: false)],
+                                mavenLocalRepo: ".repository") {
+                                sh "mvn -Djacoco.haltOnFailure=${params.failIfCoverageNotMet} -V -B clean install -e -Dmaven.test.failure.ignore=true"
+                            }
+                        }
                     }
-                }    
+                }
+            }
+        }
+    }
+        post {
+            always {
+                echo "--> We are finished with ${currentBuild.fullDisplayName}"
+                junit testResults: 'target/surefire-reports/*.xml', keepLongStdio: true
```

Comment:
maybe redundant with `   options: [junitPublisher(disabled: false)] `?

Reply:
shrug.  Not sure.

Reply:
@olamy  Please feel free to change this as needed. It will likely need further changing when ci is enabled.

Reply:
same for jacoco we can maybe use something such  `options: [junitPublisher(disabled: false), jacocoPublisher(disabled: false)]` and move jacoco configuration to pom so you have same result running build manually and running CI?

Reply:
I don't know the answer about the redundancy either. The `withMaven` step isn't one I use very often. But if it's indeed redundant, we can certainly do one or the other and not both. I'll leave that up to @olamy .

@bitwiseman is right that this will probably need some changes anyway, once CI is up and going on ci.jenkins.io.

Reply:
@olamy What is your suggested change?   I don't much about `withMaven`, what should this look like?